packages/jaeger-ui/src/model/link-patterns.js

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) 2017 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import _uniq from 'lodash/uniq';
+import { getConfigValue } from '../utils/config/get-config';
+
+const parameterRegExp = /\$\{([^{}]*)\}/;
+
+export function processTemplate(template, encodeFn) {
```

Comment:
For the string interpolation, what do you think of using something like [twitter-text/stringSupplant](https://github.com/twitter/twitter-text/blob/34dc1dd9f10e9171100cdff0cb2b7a9ed9ea2bd6/js/src/lib/stringSupplant.js)? 

So, to get the parameter names, something like:

```js
function getParamNames(str) {
  const names = [];
  str.replace(/#\{([^\}]+)\}/g, function (_, name) {
    names.push(name);
  });
  return names;
}
```

And to apply them:

```js
function stringSupplant(str, encodeFn, map) {
  return str.replace(/#\{([^}]+)\}/g, function (_, name) {
    const v = map[name];
    return v ? encodeFn(v) : '';
  });
}
```

I think this would simplify `processTemplate()` and the resulting `template()` function.

```js
template: stringSupplant.bind(null, template, endcodeFn),
```

Reply:
Your suggestion is clearly more readable, even if it requires to execute the regular expression each time the template is executed (as opposed to only once, when `processTemplate` is called). The difference in performance (if any) is probably negligible, so I have updated the PR to use your suggestion.
Do you have any preference between `#{variable}` or `${variable}` ?

Reply:
Thank you. 

I'd suggest using the `#` to differentiate from template literals.

Reply:
Ok. I have updated the PR to use `#` instead of `$`.